code/obj/item/clothing/glasses.dm

```diff
@@ -421,29 +421,25 @@
 		else
 			boutput(user, "<span class='alert'>You put on the glasses but they show no signal. The scuttlebot is likely destroyed.</span>")
 
-	attack(obj/item/W, mob/M)
-		if(istype(W, /mob/living/critter/robotic/scuttlebot))
-			var/mob/living/critter/robotic/scuttlebot/S = W
-			if (connected_scuttlebot != W)
-				boutput(M, "You try to put the goggles back into the hat but it grumps at you, not recognizing the glasses.")
-				return 1
-
-			if (istype(S, /mob/living/critter/robotic/scuttlebot/weak))
-				var/mob/living/critter/robotic/scuttlebot/weak/O = S
-				if (O.linked_hat != null)
-					O.linked_hat.set_loc(get_turf(O))
-				else
-					new /obj/item/clothing/head/det_hat/gadget(get_turf(src))
-				boutput(M, "You stuff the goggles back into the detgadget hat. It powers down with a low whirr.")
-				qdel(O)
-				qdel(src)
+	attack(mob/living/critter/robotic/scuttlebot/W, mob/M)
+		if (connected_scuttlebot != W)
```

Comment:
this is still incorrect. You are not guaranteed to be passed a mob. Look at the proc you are overriding.